feat: optimize port handling, log management, and enhance program robustness

thornboo opened the pull request:
## 📝 描述

本次合并主要包含以下关键更新：

   1. 增强的健壮性和用户体验 (Enhance robustness and UX)
       * 实现优雅关闭: 优化了退出逻辑，确保程序关闭时能彻底清理所有相关进程，防止僵尸进程残留。
       * 改进端口冲突处理: 当检测到端口冲突时，程序将发出警告而不是自动切换端口，给予用户更多控制权。
       * 增加日志轮换和清理功能: 引入了日志自动轮换机制以管理日志文件大小，并添加了手动清理日志的选项。
       * 添加可视化连接状态指示器: 在界面上增加了连接状态的视觉提示，让用户能更直观地了解当前状态。

   2. 优化端口冲突问题和日志提醒 (Optimized port conflict and log reminders)
       * 针对端口冲突场景进行了优化，并增加了对关键日志的提醒功能，提升了问题的可追溯性。
       * #41 
       * #42 

   3. 流程清理和冲突解决 (Process cleanup and conflict resolution)
       * 对进程清理逻辑进行了改进，并实现了初步的端口冲突解决方案。

## 🤔 附加信息

 - 我已经在本地充分测试，未出现端口冲突问题。
 - 所添加的部分样式（状态指示器、输出日志排版）都正常显示。
 - 优化的日志功能正常运行输出。

Copilot committed:
Initial plan

Copilot committed:
Implement port conflict resolution and improved process cleanup

Co-authored-by: kowyo <110339237+kowyo@users.noreply.github.com>

thornboo committed:
feat: optimized port conflict issues and added critical log reminders

thornboo committed:
feat: Enhance robustness and UX

- Implement graceful shutdown for complete process cleanup.

- Improve port conflict handling by warning instead of auto-switching.

- Add log rotation and a manual clear option.

- Add a visual connection status indicator.

thornboo committed:
feat: Add preview version build

thornboo committed:
fix: Fix preview version to build workflows

thornboo committed:
fix: Fix preview version to build workflows

thornboo committed:
Merge branch 'dev' of https://github.com/thornboo/hitsz-connect-verge into dev

# Conflicts:
#	.github/workflows/preview.yml

thornboo committed:
fix: Fix preview version to build workflows

kowyo commented on app/main.py:
```diff
@@ -7,11 +11,44 @@
+    if window:
+        try:
+            window.stop_connection()
+        except:
```
@thornboo 这里请考虑添加合适的 Logging，以及加入 `Exception`

kowyo commented on app/utils/connection_utils.py:
```diff
@@ -1,15 +1,169 @@
+                parts = lines[1].split()
+                if len(parts) > 0:
+                    return parts[0]  # Process name
+    except Exception:
+        pass
```
这里应该添加日志显示报错信息

kowyo commented on app/utils/connection_utils.py:
```diff
@@ -1,15 +1,169 @@
+        recent_lines = lines[-KEEP_LINES:]
+
+        # Add a rotation marker with timestamp
+        import datetime
```
import 应该放在文件头，保持和其他代码风格统一

kowyo commented on app/utils/connection_utils.py:
```diff
@@ -1,15 +1,169 @@
+        import datetime
+
+        timestamp = datetime.datetime.now().strftime("%H:%M:%S")
+        rotation_marker = f"--- 日志已轮转 ({timestamp}) - 保留最近 {KEEP_LINES} 行 ---"
```
日志应该使用英文，保持和原先代码风格统一。

kowyo commented on app/utils/connection_utils.py:
```diff
@@ -20,15 +174,24 @@ def handle_connection_finished(window):
         gc.collect()
 
-    window.status_label.setText("状态: 未连接")
+    # 使用新的状态更新方法，同时更新文本和指示器
```
注释应该使用英文，保持和原先代码风格统一。

kowyo replied:
下面的也是，就不啰嗦了

kowyo commented on app/utils/connection_utils.py:
```diff
@@ -37,6 +200,91 @@ def start_connection(window):
+                    port_conflicts.append(f"SOCKS5端口 {socks_port} 被其他程序占用")
+            used_ports.add(socks_port)
+        except ValueError:
+            append_log_with_rotation(
+                window, f"警告: SOCKS5端口配置无效: {window.socks_bind}"
+            )
+
+    if hasattr(window, "http_bind") and window.http_bind:
+        try:
+            http_port = int(window.http_bind)
+            if http_port in used_ports:
+                port_conflicts.append(f"HTTP端口 {http_port} 与SOCKS5端口冲突")
+            elif is_port_in_use("127.0.0.1", http_port):
+                process_info = get_port_process_info(http_port)
+                if process_info:
+                    if "系统进程" in process_info:
+                        # 特殊处理TIME_WAIT状态的端口
+                        append_log_with_rotation(
+                            window,
+                            f"⚠️  HTTP端口 {http_port} 处于TIME_WAIT状态，将尝试等待释放...",
+                        )
+                        if wait_for_port_release("127.0.0.1", http_port, max_wait=3):
+                            append_log_with_rotation(
+                                window, f"✅ HTTP端口 {http_port} 已释放"
+                            )
+                        else:
+                            port_conflicts.append(
+                                f"HTTP端口 {http_port} 被 {process_info} 占用"
+                            )
+                    else:
+                        port_conflicts.append(
+                            f"HTTP端口 {http_port} 被 {process_info} 程序占用"
+                        )
+                else:
+                    port_conflicts.append(f"HTTP端口 {http_port} 被其他程序占用")
+            used_ports.add(http_port)
+        except ValueError:
+            append_log_with_rotation(
+                window, f"警告: HTTP端口配置无效: {window.http_bind}"
+            )
```
@thornboo 请考虑使用一个辅助函数处理这部分重复的逻辑

kowyo commented on app/utils/connection_utils.py:
```diff
@@ -72,11 +334,17 @@ def start_connection(window):
 
     if window.http_bind:
+        append_log_with_rotation(
+            window, f"最终HTTP端口配置: 127.0.0.1:{window.http_bind}"
```
中文和英文之间添加空格，”最终 HTTP 端口配置“，其他类似情况请自查

kowyo reviewed (changes requested):
主要是一些格式的问题，日志框里的内容应该和 ZJU Connect 一样保持英文，如果单独弹出窗口，或者是状态指示板，可以使用中文，代码注释都使用英文。

中文和英文之间应该添加空格。

thornboo commented:
非常抱歉，我的本次PR并不规范，出现了很多代码风格和逻辑耦合的问题。我将在今天晚上重新PR修改之后的版本。
对于 Preview workflow ，我主要是想快速在我的fork repo的actions中获取到dev版本的exe程序，临时使用一下。这个文件不应该提交到PR中的。

thornboo committed:
fix: delete Preview workflow

thornboo committed:
Fix: Optimize comments, add loggers and exceptions to key code sections

- Change comments and logs to English
- Add logger and Exception to the critical code section
- Optimized the logical judgment of some code snippets

kowyo commented:
感谢你的 PR，我会尽快 review 修改后的代码。

kowyo commented on app/utils/connection_utils.py:
```diff
@@ -1,15 +1,249 @@
+
+        # Check if occupied by system process (TIME_WAIT)
+        process_info = get_port_process_info(port)
+        if process_info and ("系统进程" in process_info or "System process" in process_info):
```
这里应该移除中文字符

kowyo commented on app/utils/connection_utils.py:
```diff
@@ -1,15 +1,249 @@
+        elif is_port_in_use("127.0.0.1", port_value):
+            process_info = get_port_process_info(port_value)
+            if process_info:
+                if ("系统进程" in process_info) or ("System process" in process_info):
```
这里同上，应该移除中文字符

thornboo committed:
Refactor: Move cleanup and signal handling to process_utils module

- Extracted cleanup and signal handling logic from main.py to app/utils/process_utils.py
- Improved cross-platform signal registration (SIGINT, SIGTERM, SIGBREAK)
- Enhanced port checking for TIME_WAIT state on Unix-like systems
- Removed Chinese localization strings in favor of English-only logging
- Simplified main.py by delegating cleanup responsibilities to dedicated module

kowyo commented on app/main.py:
```diff
@@ -1,3 +1,6 @@
+import sys
+import logging
```
logging 和 sys 在 main.py 中都没有用到，你可以说一下为什么要在这里引入这两个 module 吗

kowyo commented on app/utils/set_proxy.py:
```diff
@@ -1,4 +1,7 @@
 import subprocess
+import os
+import signal
```
signal 没有使用

kowyo commented on app/utils/process_utils.py:
```diff
@@ -0,0 +1,38 @@
+        app and app.quit()
+        sys.exit(0)
+
+    # 注册信号处理器
```
注释改成英文

kowyo commented on app/utils/process_utils.py:
```diff
@@ -0,0 +1,38 @@
+    try:
+        window.stop_connection()
+    except (SystemExit, KeyboardInterrupt):
+        raise  # 让系统级异常正常传播
```
注释改成英文

kowyo commented on app/utils/set_proxy.py:
```diff
@@ -37,46 +40,197 @@ def __init__(self, command_args, proxy_enabled, window=None):
+                try:
+                    self.process.kill()
+                    self.process.wait(timeout=2)
+                except:
```
Do not use bare `except`

kowyo replied:
改成：
```
# stop()
try:
    self.process.kill()
    self.process.wait(timeout=2)
except Exception:
    pass
```

kowyo commented on app/utils/set_proxy.py:
```diff
@@ -37,46 +40,197 @@ def __init__(self, command_args, proxy_enabled, window=None):
+                                capture_output=True,
+                                timeout=2
+                            )
+                        except:
```
这里 except 的使用情况过于宽泛

kowyo replied:
改成
```
# _cleanup_remaining_processes()
try:
    subprocess.run(["kill", "-TERM", pid], capture_output=True, timeout=2)
except Exception:
    try:
        subprocess.run(["kill", "-KILL", pid], capture_output=True, timeout=2)
    except Exception:
        pass
```

kowyo commented on app/utils/set_proxy.py:
```diff
@@ -37,46 +40,197 @@ def __init__(self, command_args, proxy_enabled, window=None):
+                )
+            except FileNotFoundError as e:
+                self.output.emit(f"Failed to start process: executable not found\n{str(e)}\n")
+                error_occurred = True
```
error_occurred 这个变量没有被使用过

kowyo reviewed (changes requested):
我已经 review 一遍了，可以用 `uvx ruff check` 来初步检查一下是否有不规范的地方。_cleanup_remaining_processes 这里通过 `exe_name` 来杀死进程，AI 反馈这样比较不规范，如果名字是比较泛用的进程的话可能会误杀，可以用 PID 代替——这个我不太确定能不能改。

kowyo commented:
抱歉，这个 PR 代码改变的功能不止一个。当时没有及时反馈， 整体上对代码修改的逻辑不是很认同。先暂时关闭。
